### PR TITLE
client-io: simplify pargrp_iomap_init

### PR DESCRIPTION
**Pull request by andriytk**
Simplify bufs allocations at pargrp_iomap_init() with help of new pargrp_iomap_bufs_alloc() and pargrp_iomap_bufs_free() routines.

### CODE REVIEW

**Comment by andriytk** [comment]
cc @siningwuseagate

**Review by huanghua78 — APPROVED** [review]
Looks fine.

**Review by siningwuseagate — APPROVED** [review]

**Comment by huanghua78** [comment]
The mini-provisioning result shows success, but why here we have a failure result:

http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-motr/216/

**Comment by gowthamchinna** [comment]
run mini-provisioning

**Comment by gowthamchinna** [comment]
test this please

File: motr/io_pargrp.c [hunk 1]
```diff
@@ -2329,6 +2329,37 @@ static const struct pargrp_iomap_ops iomap_ops = {
...
+}
+
+static int pargrp_iomap_bufs_alloc(struct data_buf ****bufs_out,
+				   uint32_t row_nr, uint32_t col_nr)
+{
+	int row;
+	struct data_buf ***bufs;
+
+	M0_ALLOC_ARR(bufs, row_nr);
+	if (*bufs == NULL)
```
Comment (andriytk) [inline motr/io_pargrp.c:2348]: The bug is here, it should be `bufs == NULL`.

**Comment by huanghua78** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/300/

Jenkins testing triggered for latest PR.

**Comment by huanghua78** [comment]
http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Motr/300/

Jenkins testing triggered for latest PR.

**Comment by huanghua78** [comment]
> Thanks @huanghua78. I see it's failed, but I cannot figure out why. I don't see the error there, do you?

Maybe something wrong with the building machines. RE team will disable the failed machine.
Let me trigger the test again.

**Review by madhavemuri — APPROVED** [review]

**Pull request merged**